### PR TITLE
BST_Traversal_Algo

### PR DESCRIPTION
**Pull request by shaonakundu**
I have created Binary Search Tree traversal algorithms(Pre, post and inorder) in dsalgo folder using Python. Thanks for assigning me this task for Hactoberfest2020 and it was wonderful to contribute.

### CODE REVIEW

**Review by Aryamanz29 — CHANGES_REQUESTED** [review]
```
l = int(input())

arr = list(map(int, input().split()))

for i in range(l):
	tree.create(arr[i])


preorder(tree.root)
postorder(tree.root)
inorder(tree.root
```

For testing use python test framework unittest and add your test cases their only implementation of BST algo will remain in **BST_Traversal_Algorithms.py** .The path where you going to write tests for this issue is **dsalgo/tests** & named as "BST_algo_test.py". For more info regarding unittest framework visit:
[](https://docs.python.org/2/library/unittest.html)

**Comment by shaonakundu** [comment]
> ```
> l = int(input())
> 
> arr = list(map(int, input().split()))
> 
> for i in range(l):
> 	tree.create(arr[i])
> 
> 
> preorder(tree.root)
> postorder(tree.root)
> inorder(tree.root
> ```
> 
> For testing use python test framework unittest and add your test cases their only implementation of BST algo will remain in **BST_Traversal_Algorithms.py** .The path where you going to write tests for this issue is **dsalgo/tests** & named as "BST_algo_test.py". For more info regarding unittest framework visit:

Hey, I have uploaded the unittest, can you check and let me know if its correct or not

**Review by devkapilbansal — CHANGES_REQUESTED** [review]
@shaonakundu link the issue in your commit. Also, squash your commits in one. It's always better to have a single commit for an issue to track it later

**Comment by shaonakundu** [comment]
> @shaonakundu link the issue in your commit. Also, squash your commits in one. It's always better to have a single commit for an issue to track it later

Its not an issue I was asked to add another file to tests folder, so I asked to check it

**Comment by codesankalp** [comment]
@shaonakundu we will not approve your pull request until it's linked with an issue.
Linking the issue with pull request is a must step.
Also make the code changes as mengtioned by @devkapilbansal

**Comment by shaonakundu** [comment]
Ok so you mean this tree traversal issue to be linked with pull request?

**Comment by shaonakundu** [comment]
Can you guide me how to link the issue with pull req

File: dsalgo/BST_Traversal_Algorithms.py [hunk 1]
```diff
@@ -0,0 +1,98 @@
+“””  Binary Search Tree : tree where parent has two child nodes and left child node value is less than Root/parent and right child node value is greater than parent.
+
```
Comment (Aryamanz29) [inline dsalgo/BST_Traversal_Algorithms.py:2]: Please use proper docstring syntax """

File: dsalgo/BST_Traversal_Algorithms.py [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+	def __str__(self):
+		return str(self.info)
+
+class BinarySearchTree:
+	def __init__(self):
+		self.root = None
+	
+	def create(self, val):
+		 if self.root = None:
```
Comment (Aryamanz29) [inline dsalgo/BST_Traversal_Algorithms.py:28]: `if self.root==None:`

**Review by Aryamanz29 — CHANGES_REQUESTED** [review]

**Comment by Aryamanz29** [comment]
> Hey, I have uploaded the unittest, can you check and let me know if it's correct or not
Hello, after reviewing your code and locally  I have found many errors in both files (like improper indentations, no module name error in the test file, and many more). Kindly before creating any PR check your code locally and one more thing when you are using unittest framework use their inbuilt methods to check whether the test passes or not ( some methods are **assertEqual()**, **assetnotEqual()**, and many more)

**Comment by Aryamanz29** [comment]
> > Hey, I have uploaded the unittest, can you check and let me know if it's correct or not
> 
> Hello, after reviewing your code locally I have found many errors in both files (like improper indentations, no module name error in the test file, and many more).
> Kindly before creating any PR check your code locally and one more thing when you are using unittest framework use their inbuilt methods to check whether the test passes or not ( some methods are **assertEqual()**, **assetnotEqual()**, and many more)

@shaonakundu  Are you working on this issue?

**Comment by shaonakundu** [comment]
> > > Hey, I have uploaded the unittest, can you check and let me know if it's correct or not
> > 
> > 
> > Hello, after reviewing your code locally I have found many errors in both files (like improper indentations, no module name error in the test file, and many more).
> > Kindly before creating any PR check your code locally and one more thing when you are using unittest framework use their inbuilt methods to check whether the test passes or not ( some methods are **assertEqual()**, **assetnotEqual()**, and many more)
> 
> @shaonakundu Are you working on this issue?

I tried a lot but there is some issue with my system , you can unassign me, sorry for the inconvinience

**Comment by devkapilbansal** [comment]
@shaonakundu I would suggest you wait for some time. As this is almost complete we will look into this and will try to help

**Comment by shaonakundu** [comment]
> @shaonakundu I would suggest you wait for some time. As this is almost complete we will look into this and will try to help

Ok i dont know why its saying no module although its there and i tried with relative path also still and indentation error because I wrote it in correct format but still the issue is still there

**Pull request closed without merging**